src/openms/include/OpenMS/CHEMISTRY/DigestionEnzymeProtein.h

```diff
@@ -33,90 +33,62 @@
 // --------------------------------------------------------------------------
 //
 
-#ifndef OPENMS_CHEMISTRY_ENZYME_H
-#define OPENMS_CHEMISTRY_ENZYME_H
+#ifndef OPENMS_CHEMISTRY_DIGESTIONENZYMEPROTEIN_H
+#define OPENMS_CHEMISTRY_DIGESTIONENZYMEPROTEIN_H
 
-#include <OpenMS/CHEMISTRY/EmpiricalFormula.h>
-#include <OpenMS/DATASTRUCTURES/String.h>
-
-#include <iosfwd>
-#include <set>
-#include <vector>
+#include <OpenMS/CHEMISTRY/DigestionEnzyme.h>
 
 namespace OpenMS
 {
 
   /**
       @ingroup Chemistry
 
-      @brief Representation of an enzyme
-
-      This class represents enzymes.
+      @brief Representation of a digestion enzyme for proteins (protease)
   */
-  class OPENMS_DLLAPI Enzyme
+  class OPENMS_DLLAPI DigestionEnzymeProtein: public DigestionEnzyme
   {
-public:
+  public:
 
     /** @name Constructors
     */
     //@{
+    /// default constructor
+    DigestionEnzymeProtein();
+
     /// copy constructor
-    Enzyme(const Enzyme& enzyme);
+    DigestionEnzymeProtein(const DigestionEnzymeProtein& enzyme);
 
     /// detailed constructor
-    explicit Enzyme(const String& name,
-                    const String& cleavage_regex,
-                    const std::set<String>& synonyms = std::set<String>(),
-                    String regex_description = "",
-                    EmpiricalFormula n_term_gain = EmpiricalFormula("H"),
-                    EmpiricalFormula c_term_gain = EmpiricalFormula("OH"),
-                    String psi_id = "",
-                    String xtandem_id = "",
-                    UInt comet_id = 0,
-                    Int msgf_id = -1,
-                    UInt omssa_id = 0);
+    explicit DigestionEnzymeProtein(const String& name,
+                                    const String& cleavage_regex,
+                                    const std::set<String>& synonyms = std::set<String>(),
+                                    String regex_description = "",
+                                    EmpiricalFormula n_term_gain = EmpiricalFormula("H"),
+                                    EmpiricalFormula c_term_gain = EmpiricalFormula("OH"),
+                                    String psi_id = "",
+                                    String xtandem_id = "",
+                                    UInt comet_id = 0,
+                                    Int msgf_id = -1,
+                                    UInt omssa_id = 0);
 
     /// destructor
-    virtual ~Enzyme();
+    virtual ~DigestionEnzymeProtein();
     //@}
 
     /** @name Assignment
      */
     //@{
     /// assignment operator
-    Enzyme& operator=(const Enzyme& enzyme);
+    DigestionEnzymeProtein& operator=(const DigestionEnzymeProtein& enzyme);
     //@}
 
+    /// returns the substrate (here: protein)
+    enum DigestionEnzyme::Substrate getSubstrate() const;
```

Comment:
is this needed? sounds like a backdoor to dynamic type id. 
So either you know at the caller what type you have and don't need this function.
Or you use a DigestionENzyme* and don't know the type but need to deduce it using some dynamic_cast or type id. In both cases this usually indicates that there is something wrong with the design.

Reply:
I think you're right. I'll remove it.